hardware/detect.py

```diff
@@ -939,8 +939,34 @@ def clean_tuples(lst):
     return [tuple([clean_str(val) for val in elt]) for elt in lst]
 
 
-def _main(options):
-    'Command line entry point.'
+def parse_args(args):
+    """Arguments parser."""
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-H', '--human',
+                        help='Print output in human readable format',
+                        action='store_true',
+                        default=False)
+    parser.add_argument('--benchmark', '-b',
+                        choices=['cpu', 'mem', 'disk'],
+                        nargs='+',
+                        help=('Run benchmark for specific components. '
+                              'Valid components are: cpu, mem, disk'))
+    parser.add_argument('--benchmark-disk-destructive',
+                        help=('If specified make the disk component benchmark '
+                              'to be destructive'),
+                        action='store_true',
+                        default=False)
+
+    return parser.parse_args(args)
+
+
+def main():
+    """Command line entry point."""
+
+    os.environ["LANG"] = "en_US.UTF-8"
+    parser = parse_args(sys.argv[1:])
```

Comment:
nit: This isn't a parser it's args.  It's just a namespace st this point all the 'parser' aspects are handled in parse_args()

question: What is the benefit to doign the parser in you paares_args() function.  I feel like it'd be a little cleaner if it were all inline in main ... or does that impact the complexity?

Reply:
I agree, I will rename parser to args

I believe moving the parser to its own function is more pythonic and actually a cleaner approach:
- improves readability; the arguments and options are clearly segregated to the parse_args() function
- reduce complexity; main() should contain just calls to functions, not complex code
- eases the testing part; having a function dedicated to arguments is easier to test (especially for unit tests) instead of testing the entire main()